src/Stl.Interception/ServiceCollectionExt.cs

```diff
@@ -10,4 +10,60 @@ public static IServiceCollection AddTypeViewFactory(this IServiceCollection serv
         services.AddSingleton<ITypeViewFactory, TypeViewFactory>();
         return services;
     }
+
+    public static IServiceCollection AddScopedFactory<TFactory>(this IServiceCollection services) where TFactory : IMethodBasedFactory, IRequiresAsyncProxy
+    {
+        return AddFactory<TFactory>(ServiceLifetime.Scoped);
+    }
+
+    public static IServiceCollection AddScopedFactory<TFactory, TProxy>(this IServiceCollection services) 
+        where TFactory : IMethodBasedFactory, IRequiresAsyncProxy
+        where TProxy : TFactory
+    {
+        return AddFactory<TFactory, TProxy>(ServiceLifetime.Scoped);
+    }
+
+    public static IServiceCollection AddSingletonFactory<TFactory>(this IServiceCollection services) where TFactory : IMethodBasedFactory, IRequiresAsyncProxy
+    {
+        return AddFactory<TFactory>(ServiceLifetime.Singleton);
+    }
+
+    public static IServiceCollection AddSingletonFactory<TFactory, TProxy>(this IServiceCollection services) 
+        where TFactory : IMethodBasedFactory, IRequiresAsyncProxy
+        where TProxy : TFactory
+    {
+        return AddFactory<TFactory, TProxy>(ServiceLifetime.Singleton);
+    }
+
+    public static IServiceCollection AddFactory<TFactory>(this IServiceCollection services, ServiceLifetime lifetime) where TFactory : IMethodBasedFactory, IRequiresAsyncProxy
+    {
+        var factoryType = typeof(TFactory);
+        // setup-validation of existing proxy type.
+        // will fail on call to AddFactory()
+        return AddFactory(factoryType, Proxies.GetProxyType(factoryType), lifetime);
+    }
+
+    public static IServiceCollection AddFactory<TFactory, TProxy>(this IServiceCollection services, ServiceLifetime lifetime)
+        where TFactory : IMethodBasedFactory, IRequiresAsyncProxy
+        where TProxy : TFactory
+    {
+        return AddFactory(services, typeof(TFactory), typeof(TProxy), lifetime);
```

Comment:
Allows overwriting the Proxies.GetProxyType reflection method of getting the proxy type.
Used like `AddFactory<IFactory, StlInterceptionProxies.IFactoryProxy>()`.

Reply:
Got it.